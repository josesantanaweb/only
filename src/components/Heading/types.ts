export type tagType = "h1" | "h2" | "h3" | "h4" | "h5" | "h6";

export interface HeadingProps {
  /** Sets tag heading */
  className?: string;
  /** Sets tag heading */
  tag?: tagType;
  /** Sets as heading */
  as?: tagType;
  /** Sets color heading */
  variant?: "primary" | "secondary" | "tertiary";
  /** Sets text-transform heading */
  uppercase?: boolean;
}
