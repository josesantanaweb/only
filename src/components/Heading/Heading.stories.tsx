import React from "react";

import { ComponentStory, ComponentMeta } from "@storybook/react";

import Heading from "./Heading";

export default {
  title: "Heading",
  component: Heading,
  argTypes: {},
} as ComponentMeta<typeof Heading>;

const Template: ComponentStory<typeof Heading> = (args) => (
  <Heading {...args} />
);

export const Heading1 = Template.bind({});
Heading1.args = {
  tag: "h1",
  children: "Heading 1",
  variant: "tertiary",
};

export const Heading2 = Template.bind({});
Heading2.args = {
  tag: "h2",
  children: "Heading 2",
  variant: "tertiary",
};

export const Heading3 = Template.bind({});
Heading3.args = {
  tag: "h3",
  children: "Heading 3",
  variant: "tertiary",
};

export const Heading4 = Template.bind({});
Heading4.args = {
  tag: "h4",
  children: "Heading 4",
  variant: "tertiary",
};

export const Heading5 = Template.bind({});
Heading5.args = {
  tag: "h5",
  children: "Heading 5",
  variant: "tertiary",
};

export const Heading6 = Template.bind({});
Heading6.args = {
  tag: "h6",
  children: "Heading 6",
  variant: "tertiary",
};
