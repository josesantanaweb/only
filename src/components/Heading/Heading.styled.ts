import styled, { css } from "styled-components";
import { HeadingProps } from "./types";

export const HeadingStyled = styled.h6<HeadingProps>(
  ({ theme, uppercase, as, variant }) => {
    const uppercaseMixin = css`
      text-transform: ${uppercase ? "uppercase" : "capitalize"};
    `;

    const variantMixin = () => {
      switch (variant) {
        case "primary":
          return css`
            color: ${theme.colors.primary};
          `;
        case "secondary":
          return css`
            color: ${theme.colors.secondary};
          `;
        default:
          return css`
            color: ${theme.colors.tertiary};
          `;
      }
    };

    const sizeMixin = () => {
      switch (as) {
        case "h1":
          return css`
            font-size: 48px;
          `;
        case "h2":
          return css`
            font-size: 36px;
          `;
        case "h3":
          return css`
            font-size: 32px;
          `;
        case "h4":
          return css`
            font-size: 24px;
          `;
        case "h5":
          return css`
            font-size: 18px;
          `;
        case "h6":
          return css`
            font-size: 16px;
          `;
        default:
          return css`
            font-size: 16px;
          `;
      }
    };

    return css`
      display: inline-block;
      margin: 0;
      font-weight: 500;
      padding: 0;
      line-height: 1;
      ${sizeMixin};
      ${variantMixin};
      ${uppercaseMixin};
    `;
  }
);

export default HeadingStyled;
