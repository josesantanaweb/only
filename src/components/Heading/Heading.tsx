import React, { FC } from "react";
import HeadingStyled from "./Heading.styled";
import { HeadingProps } from "./types";

const Heading: FC<HeadingProps> = ({
  children,
  tag,
  className,
  uppercase = false,
  variant,
}) => (
  <HeadingStyled
    as={tag}
    className={className}
    uppercase={uppercase}
    variant={variant}
  >
    {children}
  </HeadingStyled>
);

export default Heading;
