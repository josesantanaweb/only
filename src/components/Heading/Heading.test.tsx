import React from "react";
import { render } from "../../test";
import Heading from "./Heading";

describe("Heading", () => {
  it("renders the avatar correctly", () => {
    const { container } = render(<Heading>Titulo</Heading>);
    expect(container).toMatchSnapshot();
  });

  it("should render with all the props", () => {
    const { container } = render(
      <Heading variant="primary" uppercase tag="h1" className="mb-20">
        Titulo
      </Heading>
    );
    expect(container).toMatchSnapshot();
  });
});
