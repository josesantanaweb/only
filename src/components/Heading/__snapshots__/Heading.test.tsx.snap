// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading renders the heading correctly 1`] = `
<div>
  <h6
    class="sc-bdnxRM dZLtjq"
  >
    Titulo
  </h6>
</div>
`;

exports[`Heading should render with all the props 1`] = `
<div>
  <h1
    class="sc-bdnxRM khOXzl mb-20"
  >
    Titulo
  </h1>
</div>
`;
